src/MSBuild/ValidateMSBuildPackageDependencyVersions.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Xml;
+namespace MSBuild
+{
+    public class ValidateMSBuildPackageDependencyVersions : Task
+    {
+        [Required]
+        public string AppConfig { get; set; }
+        [Required]
+        public string AssemblyPath { get; set; }
+
+        private string[] assembliesToIgnore = { "Microsoft.Build.Conversion.Core", "Microsoft.NET.StringTools.net35", "Microsoft.Build.Engine", "Microsoft.Activities.Build", "XamlBuildTask" };
```

Comment:
Comment each of these with a reason.

Reply:
Note that some of these are in this repo and we would ideally check them. The best option I can think of is to move this validation to the VSIX packaging project, but it looks like we don't actually get an expanded copy of the VSIX zip there, so that's nontrivial. It does ensure that we're checking the binding redirects we ship against the assemblies we ship though . . .

Reply:
Can you provide a justification for the last two? I don't know what they are, so I'm assuming we don't own them, so I'm assuming we shouldn't have binding redirects for them, especially since we don't ship them. I was hoping you'd say "no reason to keep them," and I would delete their binding redirects. Will add comments for the first three.